### ISSUE TITLE
Border node rendering is broken

### ISSUE THREAD

**Issue by pcdavid** [eclipse-sirius/sirius-components#359]
This looks like a regression.

### Steps to reproduce:

1. Create a "Topography" diagram on the sample "Robot" Flow model
2. Select the "Central Unit" element in the explorer and create a new object of type "Power Output" on it.
3. On the diagram, the new element should be displayed on the external border of the "Central Unit" container, with a small size. On `master` it is displayed inside the container, at the top-left corner and with a "normal" size (too big for a border node).

How is was before (on an older version of the code):

![demo-ok](https://user-images.githubusercontent.com/10608/109625833-15468a80-7b40-11eb-81c6-e47de2c8a49f.png)

How is it now on `master`:

![staging-broken](https://user-images.githubusercontent.com/10608/109625872-2099b600-7b40-11eb-8d15-6d31d728d2ab.png)

**Comment by pcdavid** [comment]
On the working version, the `borderNodes` object inside the "Central Unit" node looks like this:

```json
		"borderNodes": [
			{
				"id": "00240750-9b5e-49cb-8e46-3123a91c7c1e",
				"type": "node:image",
				"targetObjectId": "099e0d27-07a0-4069-bafb-f21dcc3ee479",
				"targetObjectKind": "flow::PowerOutput",
				"targetObjectLabel": "1000",
				"descriptionId": "f4bc113e-3605-40b9-8638-93fe7699f6ba",
				"label": {
					"id": "00240750-9b5e-49cb-8e46-3123a91c7c1e_label",
					"type": "label:inside-center",
					"text": ""
					// removed style/position/size/alignment
				},
				"style": { "imageURL": "/api/images/images/power_output.svg" },
				"position": {
					"x": -30,
					"y": 133.953125
				},
				"size": {
					"width": 30,
					"height": 30
				},
				"borderNodes": [],
				"childNodes": []
			}
		],
```

On the broken version it looks like this:

```json
	"borderNodes": [
			{
				"id": "b6dd9d36-b210-39df-a704-0a8e47bec6d9",
				"type": "node:image",
				"targetObjectId": "c5123bf3-9496-4067-818a-e5c2f3f7c742",
				"targetObjectKind": "flow::PowerOutput",
				"targetObjectLabel": "1000",
				"descriptionId": "d488ab1f-0597-43bc-bd6a-1abcfad88dff",
				"label": {
					"id": "b6dd9d36-b210-39df-a704-0a8e47bec6d9_label",
					"type": "label:inside-center",
					"text": "",
					// removed style/position/size/alignment
					"__typename": "Label"
				},
				"style": {
					"imageURL": "/api/images/images/power_output.svg",
					"__typename": "ImageNodeStyle"
				},
				"position": {
					"x": 0,
					"y": 0,
					"__typename": "Position"
				},
				"size": {
					"width": 54,
					"height": 54,
					"__typename": "Size"
				},
				"__typename": "Node",
				"borderNodes": [],
				"childNodes": []
			}
		],
```

Note the difference in `position` and `size`, which were correctly computed before but seem to get the default values now.

**Comment by pcdavid** [comment]
Simpler scenario:

1. Start from an empty model.
2. Create a Topography diagram.
3. Create a composite processor (it appears on the diagram).
4. Create a "Power Output" object inside the composite: it appears *inside* the composite processor, like a normal child element.
5. Re-create a new "Topography" on the composite processor: this one is layouted by ELK and the power output node is correctly placed on the border.

![Capture-2021-03-02_10-51-55](https://user-images.githubusercontent.com/10608/109631196-01058c00-7b46-11eb-90cc-b40f8dd27dac.png)

![Capture-2021-03-02_10-51-48](https://user-images.githubusercontent.com/10608/109631208-02cf4f80-7b46-11eb-9771-2ebe0b6a0b95.png)

**Comment by Gelio** [comment]
I have noticed that the "Arrange all" button puts border node in correct places with correct sizes. Changing anything on the diagram after that reverts to border nodes being rendered inside their parents.


https://user-images.githubusercontent.com/889383/144811502-2f20f319-8497-4e22-ae46-0146ca704ad5.mp4

**Comment by Gelio** [comment]
Hey! I see the issue has been closed, but I am still experiencing it. Border nodes are still placed inside of the parent container, not on its border


https://user-images.githubusercontent.com/889383/149199243-1201cf86-282c-4970-927c-edc36717c904.mp4

**Comment by sbegaudeau** [comment]
I don't think that this issue (border nodes acting as child nodes) will be fixed in Sirius Components 2022.01.0, only the size part. We should probably create another issue specifically for that.

**Comment by Gelio** [comment]
Good to know! I have created https://github.com/eclipse-sirius/sirius-components/issues/956 for that
